polars/polars-io/src/csv/buffer.rs

```diff
@@ -423,15 +423,11 @@ where
         return Ok(());
     };
 
-    match infer_pattern_single(val) {
-        None => {
-            buf.builder.append_null();
-            Ok(())
-        }
-        Some(pattern_with_offset) => {
-            match DatetimeInfer::<T::Native>::try_from(pattern_with_offset.pattern) {
+    match &buf.compiled {
+        Some(compiled) => {
+            match DatetimeInfer::<T::Native>::try_from(compiled.pattern_with_offset.pattern) {
```

Comment:
I was getting that `pattern_with_offset` is a private field, so I made it public - is there a way to avoid that? will take another look